src/mlpack/core/data/imputation_methods/mean_imputation.hpp

```diff
@@ -0,0 +1,198 @@
+  void Impute(const arma::Mat<T>& input,
+              arma::Mat<T>& output,
+              const T& mappedValue,
+              const size_t dimension,
+              const bool columnMajor = true)
+  {
+    // set size of the output
+    output.set_size(input.n_rows, input.n_cols);
+
+    double sum = 0;
+    size_t elems = 0; // excluding nan or missing target
+
+    using PairType = std::pair<size_t, size_t>;
+    // dimensions and indexes are saved as pairs inside this vector.
+    std::vector<PairType> targets;
+
+    // calculate number of elements and sum of them excluding mapped value or
+    // nan. while doing that, remember where mappedValue or NaN exists.
+    if (columnMajor)
+    {
+      for (size_t row = 0; row < input.n_rows; ++row)
+      {
+        for (size_t col = 0; col < input.n_cols; ++col)
+        {
+          if (row == dimension)
+          {
+            if (input(row, col) == mappedValue ||
+                std::isnan(input(row, col)))
+            {
+              targets.emplace_back(row, col);
+            }
+            else
+            {
+              elems++;
+              sum += input(row, col);
+              output(row, col) = input(row, col);
+            }
+          }
+          else
+          {
+            output(row, col) = input(row, col);
+          }
+        }
+      }
+    }
+    else
+    {
+      for (size_t col = 0; col < input.n_cols; ++col)
+      {
+        for (size_t row = 0; row < input.n_rows; ++row)
+        {
+          if (col == dimension)
+          {
+            if (input(row, col) == mappedValue ||
+                std::isnan(input(row, col)))
+            {
+              targets.emplace_back(row, col);
+            }
+            else
+            {
+              elems++;
+              sum += input(row, col);
+              output(row, col) = input(row, col);
+            }
+          }
+          else
+          {
+            output(row, col) = input(row, col);
+          }
+        }
+      }
+    }
+
+    if (elems == 0)
+      Log::Fatal << "it is impossible to calculate mean; no valid elements in "
+          << "the dimension" << std::endl;
+
+    // calculate mean;
+    const double mean = sum / elems;
+
+    // Now replace the calculated mean to the missing variables
+    // It only needs to loop through targets vector, not the whole matrix.
+    for (const PairType& target : targets)
+    {
+      output(target.first, target.second) = mean;
+    }
+  }
+
+  /**
+   * Impute function searches through the input looking for mappedValue and
+   * replaces it with the mean of the given dimension. The result is overwritten
+   * to the input matrix.
+   *
+   * @param input Matrix that contains mappedValue.
+   * @param mappedValue Value that the user wants to get rid of.
+   * @param dimension Index of the dimension of the mappedValue.
+   * @param columnMajor State of whether the input matrix is columnMajor or not.
+   */
+  void Impute(arma::Mat<T>& input,
```

Comment:
I wanted to reuse the codes of the overloads, but 
Impute(input, input, ...) searches through all values inside the matrix (to produce output), while Impute(input) searches through only with the given dimension.
It would not be very efficient. What do you think?